packages/gateway/src/v2-events/events/root-resolvers.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import { env } from '@gateway/environment'
+import { GQLResolver } from '@gateway/graphql/schema'
+import type { AppRouter } from '@opencrvs/events/src/router'
+import { createTRPCClient, httpBatchLink, HTTPHeaders } from '@trpc/client'
+
+import superjson from 'superjson'
+import uuid from 'uuid'
+
+const trpc = createTRPCClient<AppRouter>({
+  links: [
+    httpBatchLink({
+      url: env.EVENTS_URL,
+      transformer: superjson,
+      headers({ opList }) {
+        const headers = opList[0].context?.headers
```

Comment:
Not at all happy about this. opList is a non-empty list but still..

Reply:
Could we parse it and throw to be sure, have a default or acknowledge the possibly undefined?